codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddBuiltinPlugins.java

```diff
@@ -220,4 +223,15 @@ private static boolean testInputContainsMember(
     private static boolean testServiceId(Shape serviceShape, String expectedId) {
         return serviceShape.getTrait(ServiceTrait.class).map(ServiceTrait::getSdkId).orElse("").equals(expectedId);
     }
+
+    private static boolean operationUsesAwsAuth(Model model, ServiceShape service, OperationShape operation) {
+        // Cognito Identity service doesn't need auth for GetId, GetOpenIdToken, GetCredentialsForIdentity.
+        if (testServiceId(service, "Cognito Identity")
+                && SetUtils.of("GetId", "GetOpenIdToken", "GetCredentialsForIdentity")
+                .contains(operation.getId().getName())
```

Comment:
Nit: Can this be indented 8 more spaces? That'll help indicate it's still part of the previous line's statement.